src/algebra/quaternion.lean

```diff
@@ -84,12 +84,21 @@ def equiv_tuple {R : Type*} (c₁ c₂ : R) : ℍ[R, c₁, c₂] ≃ (fin 4 →
 
 variables {S T R : Type*} [comm_ring R] {c₁ c₂ : R} (r x y z : R) (a b c : ℍ[R, c₁, c₂])
 
+/-- The imaginary part of a quaternion. -/
+def im (x : ℍ[R, c₁, c₂]) : ℍ[R, c₁, c₂] := ⟨0, x.im_i, x.im_j, x.im_k⟩
+
+@[simp] lemma im_re : a.im.re = 0 := rfl
+@[simp] lemma im_im_i : a.im.im_i = a.im_i := rfl
+@[simp] lemma im_im_j : a.im.im_j = a.im_j := rfl
+@[simp] lemma im_im_k : a.im.im_k = a.im_k := rfl
+@[simp] lemma im_im : a.im.im = a.im := rfl
```

Comment:
```suggestion
@[simp] lemma im_idem : a.im.im = a.im := rfl
```

Reply:
This seems worse to me because it doesn't match the pattern of the three lemmas above.

Reply:
Yes, but this is an idempotency lemma. And all idempotency lemmas I know end in `_idem`. See `convex_hull_idem`, `intrinsic_closure_idem`...

At any rate, your name happens to be a substring of mine, so it can't hurt.

Reply:
Alright, changed